src/firewall/server/decorators.py

```diff
@@ -137,7 +140,9 @@ def __call__(self, func):
         )
         @functools.wraps(func)
         def _impl(*args, **kwargs):
-            pass
+            GLib.idle_add(
```

Comment:
it doesn't seem right that this dispatcher runs for *every* D-Bus signal.

There are *a lot* of signals, and having a dispatcher for all of them seems very expensive, but also unnecessary.

It's also not clear that this should be tied to D-Bus signals at all. Just define events and points in time when a dispatcher script is invoked, and explicitly run it at that point. Yes, sometimes a D-Bus signal and a dispatcher script will be invoked together, but most often not.

E.g. the `PropertiesChanged` signal seems not a useful signal for also calling a dispatcher. But aside from a few examples where it's not useful, it's mainly about how expensive it is to spawn that many processes, over and over. And commiting to do such expensive actions that in form of a new API.

Reply:
> it doesn't seem right that this dispatcher runs for every D-Bus signal.

I could see omitting some stuff, e.g. `PropertiesChanged`, that you point out below, but I think we want to see all runtime and permanent configuration changes.

> It's also not clear that this should be tied to D-Bus signals at all. 

Both dbus signals and dispatcher are a "something interesting" happened. Why maintain two different concepts or sets of "events"? Reusing the dbus signals gives us dispatchers for cheap.

> But aside from a few examples where it's not useful, it's mainly about how expensive it is to spawn that many processes, over and over. And commiting to do such expensive actions that in form of a new API.

This costs only exists if there are dispatchers to call.